index.js

```diff
@@ -0,0 +1,114 @@
+import assert from 'node:assert'
+import ms from 'ms'
+
+export class CancelStuckTransactions {
+  #store
+  #list
+  #resolve
+  #log
+  #sendTransaction
+  constructor ({ store, list, resolve, log, sendTransaction }) {
+    this.#store = store
+    this.#list = list
+    this.#resolve = resolve
+    this.#log = log
+    this.#sendTransaction = sendTransaction
+  }
+
+  async pending (tx) {
+    assert.strictEqual(typeof tx.hash, 'string')
+    assert.strictEqual(typeof tx.from, 'string')
+    assert.strictEqual(typeof tx.maxPriorityFeePerGas, 'bigint')
+    assert.strictEqual(typeof tx.nonce, 'number')
+    await this.#store({
+      ...tx,
+      timestamp: new Date().toISOString()
+    })
+  }
+
+  async successful (tx) {
+    assert.strictEqual(typeof tx.hash, 'string')
+    await this.#resolve(tx.hash)
+  }
+
+  async olderThan (ageMs) {
```

Comment:
I find the naming of these methods confusing.

Some are dealing with the status of individual transactions (pending, successful), this one performs a scan + replace action, which is not clear from the description.

Methods & functions usually start with a verb.

Proposed names:

```
addPending(tx)
removeSuccessful(tx)
cancelAllOlderThan(ageMs)
```

Reply:
I was writing this "magic" style where it reads more like spoken word. I've changed it to be explicit